### PR TITLE
tests: net: getaddrinfo: Disable onboard Ethernet controller

### LINKED ISSUE #34923: net.socket.get_addr_info: frdm_k64f test fails
**Describe the bug**
below two cases fails, looks like it is not stable enough to all platform
test_getaddrinfo_ok
test_getaddrinfo_cancelled 

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=frdm_k64f
3. make falsh

**Expected behavior**
all test pass

**Impact**
addressinfo api call

**Logs and console output**

```
    start application at 0x70008729 !!!

    start stack at 0x80008800 !!!

    *** Booting Zephyr OS build zephyr-v2.5.0-3077-g3a7e0f875cf1  ***

    Running test suite socket_getaddrinfo

    ===================================================================

    START - test_getaddrinfo_setup

     PASS - test_getaddrinfo_setup in 0.90 seconds

    ===================================================================

    START - test_getaddrinfo_ok

    

        Assertion failed at WEST_TOPDIR/zephyr/tests/net/socket/getaddrinfo/src/main.c:232: test_getaddrinfo_ok: (false is false)

    Did not receive all queries

     FAIL - test_getaddrinfo_ok in 2.155 seconds

    ===================================================================

    START - test_getaddrinfo_cancelled

    

        Assertion failed at WEST_TOPDIR/zephyr/tests/net/socket/getaddrinfo/src/main.c:252: test_getaddrinfo_cancelled: (false is false)

    Did not receive all queries

     FAIL - test_getaddrinfo_cancelled in 0.15 seconds

    ===================================================================

    START - test_getaddrinfo_no_host

     PASS - test_getaddrinfo_no_host in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_num_ipv4

     PASS - test_getaddrinfo_num_ipv4 in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_num_ipv6

     PASS - test_getaddrinfo_num_ipv6 in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_flags_numerichost

     PASS - test_getaddrinfo_flags_numerichost in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_ipv4_hints_ipv6

     PASS - test_getaddrinfo_ipv4_hints_ipv6 in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_ipv6_hints_ipv4

     PASS - test_getaddrinfo_ipv6_hints_ipv4 in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_port_invalid

     PASS - test_getaddrinfo_port_invalid in 0.1 seconds

    ===================================================================

    START - test_getaddrinfo_null_host

     PASS - test_getaddrinfo_null_host in 0.1 seconds

    ===================================================================

    Test suite socket_getaddrinfo failed.

    ===================================================================

    PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: zephyr-v2.5.0-3077-g3a7e0f875cf1

### PR DESCRIPTION
**Pull request by jukkar**
If the tests/net/socket/getaddrinfo is run in a board that has
Ethernet controller, then it is possible that it will interfere
the test if the Ethernet cable is connected. As the test only
needs loopback support to run, disable the Ethernet as it is
not needed by the test.

Fixes #34923

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by dleach02** [comment]
Maybe it is time to walk through all the net samples and fix the self hosted tests to ensure they avoid using the actual ethernet?

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
It also feels like these general sets of tests are hardware agnostic if they are not going to use the device's hardware so maybe we change the filter to just run on QEMU

**Comment by jukkar** [comment]
> It also feels like these general sets of tests are hardware agnostic if they are not going to use the device's hardware so maybe we change the filter to just run on QEMU

I fully agree, and it was like that at some point (see commit a81aad9756fc9078f102ee64bbf19583892f3dcf where tests were run for native_posix and/or qemu only). Not sure why it was changed and why the filters are like this atm.

File: tests/net/socket/getaddrinfo/prj.conf [hunk 1]
```diff
@@ -9,6 +9,7 @@ CONFIG_NET_IPV4=y
 CONFIG_NET_IPV6=y
 CONFIG_NET_SOCKETS=y
 CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_NET_L2_ETHERNET=n
```
[the remark this answers is not in the document]
Comment (jukkar) [reply tests/net/socket/getaddrinfo/prj.conf:13]: Not sure if it is worth mentioning this just for this one specific test. In fact all the tests under `tests/net` are like this, they are self contained and do not need any specific hw support to run.

**Review by pfalcon — APPROVED** [review]

**Pull request merged**